pinlockview/src/main/java/com/andrognito/pinlockview/InputField.java

```diff
@@ -0,0 +1,199 @@
+package com.andrognito.pinlockview;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Build;
+import android.os.Handler;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.AppCompatEditText;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.KeyEvent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.TextView;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class InputField extends AppCompatEditText {
+
+    private static final char PASSWORD_DOT = '\u2022';
+    private static final int DEFAULT_DOT_DELAY_MILLIS = 1500;
```

Comment:
Whats this delay for ?

Reply:
We delay the last character from turning into a dot for 1.5 seconds, after 1.5 seconds are up we turn it into a dot.